rust-lang/cargo
# fix: clean up for deprecated and removed commands

### What does this PR try to resolve?

This PR is a follow-up of <https://github.com/rust-lang/cargo/issues/14679>.

* A new doc page for deprecated and removed commands
* Remove shell completions for deprecated commands
* Remove `verify-project` and `read-manifest` from `cargo --list` output.

### How should we test and review this PR?

`mdbook serve src/doc` and read the doc.

I've set up redirections for all removed/deprecated commands to maintain consistency,
though only `verify-project` really need it I believe.
